### PR TITLE
feat: fix automatic table creation error

### LINKED ISSUE #4: AttributeError: 'AsyncEngine' object has no attribute '_run_ddl_visitor'
Please note that this code will raise an error：AttributeError

https://github.com/pycasbin/async-sqlalchemy-adapter/blob/7ad2de0676c258bf36cc29fb45069f5684c4f684/casbin_async_sqlalchemy_adapter/adapter.py#L84

The correct way to open in asynchronous should be:
```python
async with engine.begin() as conn:
        await conn.run_sync(Base.metadata.drop_all)
```

For detailed examples, please see：https://docs.sqlalchemy.org/en/14/_modules/examples/asyncio/async_orm.html

### PR DESCRIPTION
**Pull request by wu-clan**
Fix: https://github.com/pycasbin/async-sqlalchemy-adapter/issues/4

Contains some other fixes: version dependencies, testing...

### CODE REVIEW

**Comment by hsluoyz** [comment]
@wu-clan fix all:

image

**Comment by wu-clan** [comment]
@hsluoyz The semantic detection doesn't seem to pass because I didn't comply in my previous commit, and even though I reworked the upload, it doesn't overwrite the commit, so I guess can ignore the detection and merge it into the master branch by `squash and merge`

**Comment by wu-clan** [comment]
@hsluoyz please review.

File: casbin_async_sqlalchemy_adapter/adapter.py [hunk 1]
```diff
@@ -62,14 +62,20 @@ class Filter:
...
-    def __init__(self, engine, db_class=None, filtered=False):
+    def __init__(self, engine, db_class=None, warning=True, filtered=False):
         if isinstance(engine, str):
             self._engine = create_async_engine(engine, future=True)
         else:
             self._engine = engine
 
         if db_class is None:
             db_class = CasbinRule
+            if warning:
```
Comment (hsluoyz) [inline casbin_async_sqlalchemy_adapter/adapter.py:73]: Add to README, not add warning

File: casbin_async_sqlalchemy_adapter/adapter.py [hunk 2]
```diff
@@ -95,6 +100,11 @@ async def _session_scope(self):
                 await session.rollback()
                 raise e
 
+    async def create_table(self):
```
Comment (hsluoyz) [inline casbin_async_sqlalchemy_adapter/adapter.py:103]: See other adapters (like Go, Java) about how to create table. This is not Casbin-like

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: casbin_async_sqlalchemy_adapter/adapter.py (hunk 1 above)
Comment (wu-clan) [reply casbin_async_sqlalchemy_adapter/adapter.py:73]: This requires a modification to Simple Example and must be wrapped with asynchronous functions that do not look like traditional casbin

The example will look like:
```python
import casbin_async_sqlalchemy_adapter
import casbin

async def enforce():
    adapter = casbin_async_sqlalchemy_adapter.Adapter('sqlite:///test.db')
    
    # If use default CasbinRule table, please note the use of the "Adapter().create_table()"
    # method to create the table structure, and ignore this if you are using a custom CasbinRule table.
    await adapter.create_table()
    
    e = casbin.Enforcer('path/to/model.conf', adapter)
    
    sub = "alice"  # the user that wants to access a resource.
    obj = "data1"  # the resource that is going to be accessed.
    act = "read"  # the operation that the user performs on the resource.
    
    if e.enforce(sub, obj, act):
        # permit alice to read data1
        pass
    else:
        # deny the request, show an error
        pass
```

File: casbin_async_sqlalchemy_adapter/adapter.py (hunk 2 above)
Comment (wu-clan) [reply casbin_async_sqlalchemy_adapter/adapter.py:103]: I know it's abrupt, and as I mentioned in the question, you can't call asynchronous code directly in python's `__init__()` method, so it's forced.

Unless, of course, there is a better implementation @Wrapping-2000

File: casbin_async_sqlalchemy_adapter/adapter.py [hunk 3]
```diff
@@ -62,14 +62,20 @@ class Filter:
 class Adapter(persist.Adapter):
     """the interface for Casbin adapters."""
 
-    def __init__(self, engine, db_class=None, filtered=False):
+    def __init__(self, engine, db_class=None, warning=True, filtered=False):
```
Comment (hsluoyz) [inline casbin_async_sqlalchemy_adapter/adapter.py]: warning should be 4th arg?

**Review by hsluoyz — CHANGES_REQUESTED** [review]

File: casbin_async_sqlalchemy_adapter/adapter.py (hunk 3 above)
Comment (wu-clan) [reply casbin_async_sqlalchemy_adapter/adapter.py]: done.

**Review by hsluoyz — APPROVED** [review]

**Pull request merged**